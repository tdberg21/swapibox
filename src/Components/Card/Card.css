.card {
  border: 1px solid black;
  margin: 8px;
  padding-left: 5px;
  padding-right: 5px;
  padding-bottom: 10px;
  width: 29%;
  background-color: lightgrey;
}

.favorite-button {
  border-color: yellow;
  color: yellow;
  background-color: #333;
}

.favorite {
  background-color: chartreuse;
}

ul {
  list-style-type: none;
  padding-left: 0;
}

li {
  text-overflow: ellipsis;
  padding-right: 5px;
}