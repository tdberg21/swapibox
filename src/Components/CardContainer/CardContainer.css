.card-container {
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
  overflow: scroll;
  padding-right: 25px;
  padding-top: 30px;
}