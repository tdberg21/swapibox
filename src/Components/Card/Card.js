import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';


const Card = (props) => {

  let cardToRender = Object.keys(props.info).map(item => {
    console.log(props.info[item])
    return <p> {[item]} : {props.info[item]} </p>
  });

  return (
    <div className="card">
      {cardToRender}
      <button
        onClick={(event) => props.checkForFaves(props.info.id, event)}
        className="favorite-button">
          Favorite
      </button>
    </div>
  );


  // if (props.info.species) {
  //   return (
  //     <div className="card">
  //       <h5>{props.info.name}</h5>
  //       <p>Species: {props.info.species} </p>
  //       <p>Homeworld: {props.info.homeworld}</p>
  //       <p>Homeworld population:{props.info.population}</p>
  //       <button 
  //         onClick={(event) => props.checkForFaves(props.info.id, event)}
  //         className="favorite-button">
  //         Favorite
  //       </button>
  //     </div>
  //   );
  // } 
  
  // if (props.info.terrain) {
  //   const residentsToDisplay = props.info.residents.map((resident, index) => {
  //     return (<li key={index}>{resident}</li>);
  //   });
    
  //   return (
  //     <div className="card">
  //       <h5>{props.info.name}</h5>
  //       <p>Climate: {props.info.climate} </p>
  //       <p>Terrain: {props.info.terrain}</p>
  //       <p>Population:{props.info.population}</p>
  //       <ul>Residents:{residentsToDisplay}</ul>
  //       <button 
  //         onClick={(event) => props.checkForFaves(props.info.id, event)}
  //         className="favorite-button"
  //       >
  //       Favorite
  //       </button>
  //     </div>
  //   );
  // } 
   
  // if (props.info.model) {
  //   return (
  //     <div className="card">
  //       <h5>{props.info.name}</h5>
  //       <p>Model: {props.info.model} </p>
  //       <p>Class: {props.info.class}</p>
  //       <p>Passengers:{props.info.passengers}</p>
  //       <button 
  //         onClick={(event) => props.checkForFaves(props.info.id, event)}
  //         className="favorite-button"
  //       >
  //       Favorite
  //       </button>
  //     </div>
  //   );
  // }
};

Card.propTypes = {
  name: PropTypes.string,
  species: PropTypes.string,
  homeWorld: PropTypes.string,
  homePop: PropTypes.number,
  info: PropTypes.object,
  checkForFaves: PropTypes.func
};

export default Card;
