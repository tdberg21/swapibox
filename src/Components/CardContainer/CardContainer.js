import React from 'react';
import Card from '../Card/Card.js';
import './CardContainer.css';
import PropTypes from 'prop-types';

const CardContainer = ({ cards, checkForFaves }) => {
  const cardsToRender = cards.map(item => {
    return (
      <Card 
        info={item}
        key={item.id}
        checkForFaves={checkForFaves} 
      />
    );
  });
  
  return (
    <div 
      className="card-container">
      {cardsToRender}
    </div>
  );
};

CardContainer.propTypes = {
  cards: PropTypes.array,
  checkForFaves: PropTypes.func
};

export default CardContainer;