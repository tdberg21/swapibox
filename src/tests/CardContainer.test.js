import React from 'react';
import CardContainer from '../Components/CardContainer/CardContainer.js';
import Card from '../Components/Card/Card.js';
import { shallow } from 'enzyme';


describe('CARD CONTAINER TESTS', () => {
  let wrapper;
  let mockCards = [{ class: "wheeled", id: 0, model: "Digger Crawler", name: "Sand Crawler", passengers: "30" }, { class: "repulsorcraft", id: 1, model: "T-16 skyhopper", name: "T-16 skyhopper", passengers: "1" }]

  beforeEach(() => {
    wrapper = shallow(<CardContainer cards={mockCards}/>);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders all the cards', () => {
    expect(wrapper.find(Card).length).toBe(2);
  })
});