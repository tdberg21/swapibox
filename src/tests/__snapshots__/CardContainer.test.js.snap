// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD CONTAINER TESTS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cards={
        Array [
            Object {
              "class": "wheeled",
              "id": 0,
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
            },
            Object {
              "class": "repulsorcraft",
              "id": 1,
              "model": "T-16 skyhopper",
              "name": "T-16 skyhopper",
              "passengers": "1",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          checkForFaves={undefined}
          info={
                    Object {
                              "class": "wheeled",
                              "id": 0,
                              "model": "Digger Crawler",
                              "name": "Sand Crawler",
                              "passengers": "30",
                            }
          }
/>,
        <Card
          checkForFaves={undefined}
          info={
                    Object {
                              "class": "repulsorcraft",
                              "id": 1,
                              "model": "T-16 skyhopper",
                              "name": "T-16 skyhopper",
                              "passengers": "1",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "checkForFaves": undefined,
          "info": Object {
            "class": "wheeled",
            "id": 0,
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": "30",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "checkForFaves": undefined,
          "info": Object {
            "class": "repulsorcraft",
            "id": 1,
            "model": "T-16 skyhopper",
            "name": "T-16 skyhopper",
            "passengers": "1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            checkForFaves={undefined}
            info={
                        Object {
                                    "class": "wheeled",
                                    "id": 0,
                                    "model": "Digger Crawler",
                                    "name": "Sand Crawler",
                                    "passengers": "30",
                                  }
            }
/>,
          <Card
            checkForFaves={undefined}
            info={
                        Object {
                                    "class": "repulsorcraft",
                                    "id": 1,
                                    "model": "T-16 skyhopper",
                                    "name": "T-16 skyhopper",
                                    "passengers": "1",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "checkForFaves": undefined,
            "info": Object {
              "class": "wheeled",
              "id": 0,
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "checkForFaves": undefined,
            "info": Object {
              "class": "repulsorcraft",
              "id": 1,
              "model": "T-16 skyhopper",
              "name": "T-16 skyhopper",
              "passengers": "1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
